test/e2e/healthcheck_run_test.go

```diff
@@ -404,4 +404,13 @@ HEALTHCHECK CMD ls -l / 2>&1`, ALPINE)
 		Expect(ps.OutputToStringArray()).To(HaveLen(2))
 		Expect(ps.OutputToString()).To(ContainSubstring("hc"))
 	})
+
+	It("podman healthcheck - health timeout", func() {
+		ctrName := "c-h-" + RandomString(6)
+		podmanTest.PodmanExitCleanly("run", "-d", "--name", ctrName, "--health-cmd", "sleep 100; echo hc-done", "--health-timeout=3s", ALPINE, "top")
```

Comment:
In order to test the grace period and SIGTERM behavior you likely should install a signal handler to test we get the proper signal
"trap 'echo Received SIGTERM, finishing; exit' SIGTERM; while :; do sleep 1; done"

And you could do a version without the `exit` to force the SIGKILL as it doesn't exit after SIGTERM then.